### PR TITLE
Issue policy and template tweaks

### PR DESCRIPTION
**Pull request by dbczumar**
## What changes are proposed in this pull request?

* Revamp "feature request" template to prompt filers for specific motivation questions and high-level implementation details (if possible)
* Revamp Issue Policy to address the lifecycle for each issue type
* Add "Willingness to contribute" sections to the feature request, bug report, and documentation fix templates

## How is this patch tested?

Manually - I recommend viewing the associated files in a markdown renderer (e.g., open the file in the GitHub file browser).

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Model Registry
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/feature_request_template.md [hunk 1]
```diff
@@ -4,29 +4,81 @@ about: Use this template for feature and enhancement proposals.
...
+- [ ] REST-API
+- [ ] Examples
+- [ ] Docs
+- [ ] Tracking
+- [ ] Projects
+- [ ] Artifacts
+- [ ] Models
+- [ ] Model Registry
+- [ ] Scoring
+- [ ] Serving
```
Comment (dbczumar) [inline .github/ISSUE_TEMPLATE/feature_request_template.md:57]: @sueann Can we dedupe `Scoring` and `Serving` here and in the pull request template?

File: .github/ISSUE_TEMPLATE/bug_report_template.md [hunk 2]
```diff
@@ -5,9 +5,17 @@ labels: 'bug'
 title: "[BUG]"
 ---
 Thank you for submitting an issue. Please refer to our [issue policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)
-for information on what types of issues we address. For help with debugging your code, please refer to [Stack Overflow](https://stackoverflow.com/questions/tagged/mlflow).
+for additional information about bug reports. For help with debugging your code, please refer to [Stack Overflow](https://stackoverflow.com/questions/tagged/mlflow).
 
-Please fill in this template and do not delete it unless you are sure your issue is outside its scope.
+**Please fill in this bug report template to ensure a thorough response.**
```
Comment (aarondav) [inline .github/ISSUE_TEMPLATE/bug_report_template.md:10]: Do I want a _thorough_ response? Quick? Some other words?

File: .github/ISSUE_TEMPLATE/bug_report_template.md [hunk 3]
```diff
@@ -5,9 +5,17 @@ labels: 'bug'
...
+
+### Willingness to contribute
+The MLflow Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix
+for this bug to the MLflow code base?
+
+- [ ] Yes. I can contribute a fix for this bug independently.
+- [ ] Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.
+- [ ] No. **Explanation:**
```
Comment (aarondav) [inline .github/ISSUE_TEMPLATE/bug_report_template.md:18]: Probably we can leave out the Explanation bit. Just having "No" here is already 🤨

File: .github/ISSUE_TEMPLATE/feature_request_template.md [hunk 4]
```diff
@@ -4,29 +4,81 @@ about: Use this template for feature and enhancement proposals.
...
-### Proposed Changes
-For user-facing changes, what APIs are you proposing to add or modify? For internal changes, what code paths will need to be modified?
+### Can this feature be introduced as an MLflow Plugin?
+[MLflow Plugins] enable integration of third-party modules with many of MLflow’s components, allowing you to maintain and iterate on certain features independently of the MLflow Repository.
+
+Please refer to the https://mlflow.org/docs/latest/plugins.html for a list of supported MLflow Plugin types. Afterwards, select the option that best applies and fill in the **Explanation** field if applicable:
+
+- [ ] Yes. I think this feature can be introduced as an MLflow Plugin.
+- [ ] No. It does not make sense to structure this feature as an MLflow Plugin. **Explanation:**
+- [ ] No. I don’t think MLflow Plugins currently supports this type of feature. **Explanation:**
```
Comment (aarondav) [inline .github/ISSUE_TEMPLATE/feature_request_template.md:37]: s/supports/support/?

File: .github/ISSUE_TEMPLATE/feature_request_template.md (hunk 1 above)
Comment (aarondav) [reply .github/ISSUE_TEMPLATE/feature_request_template.md:57]: Maybe also API and REST-API? Dunno how a user would choose between those.

Comment (sueann) [reply .github/ISSUE_TEMPLATE/feature_request_template.md:57]: sounds good - i think we want to just keep "Scoring"

File: .github/ISSUE_TEMPLATE/bug_report_template.md (hunk 2 above)
Comment (dbczumar) [reply .github/ISSUE_TEMPLATE/bug_report_template.md]: I've changed this to `timely and thorough`. More than happy to iterate here though!

File: .github/ISSUE_TEMPLATE/bug_report_template.md (hunk 3 above)
Comment (dbczumar) [reply .github/ISSUE_TEMPLATE/bug_report_template.md]: Done!

File: .github/ISSUE_TEMPLATE/feature_request_template.md (hunk 4 above)
Comment (dbczumar) [reply .github/ISSUE_TEMPLATE/feature_request_template.md]: Done! Thanks!

File: .github/ISSUE_TEMPLATE/feature_request_template.md (hunk 1 above)
Comment (dbczumar) [reply .github/ISSUE_TEMPLATE/feature_request_template.md]: Thanks for the input! I've consolidated `Serving` and `Scoring` to just `Scoring`. I've consolidated `API` and `REST-API` as just `API`.

File: .github/ISSUE_TEMPLATE/bug_report_template.md (hunk 2 above)
Comment (dbczumar) [reply .github/ISSUE_TEMPLATE/bug_report_template.md]: ^ I applied these changes to the PR template as well

**Review by aarondav — APPROVED** [review]

**Comment by dbczumar** [comment]
@aarondav @smurching @mateiz In response to feedback that the feature request template is too lengthy and technical, I've removed the sections related to plugins and `implementation impact` from the template. Instead, I've updated the `CONTRIBUTING.rst` document to provide guidelines about feature / patch design and information about structuring features as plugins, as well as an overview of the contribution process. The feature request template explicitly encourages users to consult the contributing guide. Another round of reviews would be greatly appreciated!

File: CONTRIBUTING.rst [hunk 5]
```diff
@@ -1,13 +1,105 @@
...
+####################
+The MLflow contribution process starts with filing a GitHub issue. MLflow defines four
+categories of issues: feature requests, bug reports, documentation fixes, and installation issues.
+Details about each issue type and the issue lifecycle are discussed in the `MLflow Issue Policy
+<https://github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md>`_.
+
+MLflow committers actively triage and respond to GitHub issues. In general, we recommend waiting
+for feebdack from an MLflow committer or community member before proceeding to implement a feature
+or patch. This is particularly important for
+`significant changes <https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst#write-designs-for-significant-changes>`_.
```
Comment (dbczumar) [inline CONTRIBUTING.rst:20]: I tried using RST references (`:ref:`) to link to sections within this document. Unfortunately, GitHub doesn't seem to render RST references properly (see https://github.com/isaacs/github/issues/892), so I gave this a shot instead.

File: .github/ISSUE_TEMPLATE/feature_request_template.md [hunk 6]
```diff
@@ -4,29 +4,48 @@ about: Use this template for feature and enhancement proposals.
...
+## Motivation
+- What is the use case for this feature?
+- Why is this use case valuable to support for MLflow users in general?
+- Why is this use case valuable to support for your project(s) or organization?
+- Why is it currently difficult to achieve this use case? (please be as specific as possible about why related MLflow features and components are insufficient)
+
+## Which MLflow component(s) does this feature affect?
+
+- [ ] UI
+- [ ] CLI
```
Comment (mateiz) [inline .github/ISSUE_TEMPLATE/feature_request_template.md:33]: Some people might not know what CLI stands for so maybe spell it out

Comment (dbczumar) [reply .github/ISSUE_TEMPLATE/feature_request_template.md]: Changed this to read `Command Line Interface` and also made the corresponding change to the pull request template. Thanks for pointing this out!

File: .github/ISSUE_TEMPLATE/feature_request_template.md [hunk 7]
```diff
@@ -4,29 +4,48 @@ about: Use this template for feature and enhancement proposals.
...
-* Are extensible (e.g. if adding an integration with an ML framework, is it possible to add similar integrations with other frameworks?)
-* Have user impact & value that justifies the maintenance burden of supporting the feature moving forwards. The [JQuery contributor guide](https://contribute.jquery.org/open-source/#contributing-something-new) has an excellent discussion on this.
-
-### Motivation
-What is the use case in mind?  Why is it valuable to support, and why is it currently difficult or impossible to achieve? Could the desired functionality alternatively be implemented as a third-party package using MLflow public APIs?
-
-### Proposed Changes
-For user-facing changes, what APIs are you proposing to add or modify? For internal changes, what code paths will need to be modified?
+Thank you for submitting a feature request. **Before proceeding, please review MLflow's
+[Issue Policy for feature requests](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md##feature-requests)
```
Comment (smurching) [inline .github/ISSUE_TEMPLATE/feature_request_template.md:8]:
```suggestion
[Issue Policy for feature requests](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md#feature-requests)
```

Tiny nit, should this be a single "#"?  e.g. https://github.com/mlflow/mlflow/blob/master/CHANGELOG.rst#172-2020-03-20 works for me on chrome (jumps to 1.7.2 changelog) but https://github.com/mlflow/mlflow/blob/master/CHANGELOG.rst##172-2020-03-20 doesn't seem to work

**Review by smurching — APPROVED** [review]
One nit, but otherwise LGTM thanks @dbczumar!

File: .github/ISSUE_TEMPLATE/feature_request_template.md (hunk 7 above)
Comment (dbczumar) [reply .github/ISSUE_TEMPLATE/feature_request_template.md:8]: Yes! Thank you. Great catch!

**Pull request merged**